parser/sql/dialect/opengauss/src/main/java/org/apache/shardingsphere/sql/parser/opengauss/visitor/statement/OpenGaussStatementVisitor.java

```diff
@@ -1016,9 +1018,22 @@ public ASTNode visitHavingClause(final HavingClauseContext ctx) {
     
     @Override
     public ASTNode visitWindowClause(final WindowClauseContext ctx) {
+        if (null != ctx.windowDefinitionList()) {
+            return new WindowSegment(ctx.getStart().getStartIndex(), ctx.getStop().getStopIndex(), getWindowItem(ctx.windowDefinitionList().windowDefinition()),
+                    getWindowSpecification(ctx.windowDefinitionList().windowDefinition().windowSpecification()));
+        }
         return new WindowSegment(ctx.getStart().getStartIndex(), ctx.getStop().getStopIndex());
     }
     
+    private IdentifierValue getWindowItem(final WindowDefinitionContext ctx) {
+        return new IdentifierValue(ctx.colId().identifier().getText());
+    }
+    
+    private Collection<ExpressionSegment> getWindowSpecification(final WindowSpecificationContext ctx) {
+        Collection<ExpressionSegment> result = createInsertValuesSegments(ctx.partitionClause().exprList());
```

Comment:
Hi @kanha-gupta, why call createInsertValuesSegments method in getWindowSpecification?

Reply:
Sir, thank you for the review :) createInsertValuesSegments method Gives desired Result for exprList() processing therefore I call that method instead of writing code for it.
Is the approach good or Should I write new code for it ?
Thank you